### ISSUE TITLE
Faulty Money comparisons for negative values

### ISSUE THREAD

**Issue by UlrichEckhardt** [moneyphp/money#503]
Calculator defines an interface to compare numbers:
```
    /**
     * Compare a to b.
     *
     * @param string $a
     * @param string $b
     *
     * @return int
     */
    public function compare($a, $b);
```
Since nothing is documented concerning the meaning of the returnvalue, I would expect that it returns a negative number when `$a < $b`, zero when `$a = $b` and a positive number when `$a > $b`. This expectation conforms to e.g. http://php.net/manual/en/function.usort.php or the `<=>` operator of PHP7.

Looking at Money.php, you will find several operations where the result of `compare()` is compared to either +1 or -1, which is subtly different from above assumption. For GMP, check this example:
```
$a = gmp_init('-100');
$b = gmp_init('100');
$res = gmp_cmp($a, $b);
var_dump($res);
```
On my machine, the output is `-2`, which then breaks `Money` when comparing negative amounts.

I'm in the process of preparing a patch with the following features:
 * No more comparisons with +1 or -1 are left in `Money`.
 * Tests for comparisons of negative numbers exist.
 * Tests (CalculatorTestCase.php) don't assume a specific negative or positive value either.

**Comment by UlrichEckhardt** [comment]
Here's the promised patch:
[0001-Fix-numeric-comparison-for-negative-numbers.patch.txt](https://github.com/moneyphp/money/files/2410985/0001-Fix-numeric-comparison-for-negative-numbers.patch.txt)

**Comment by sagikazarmark** [comment]
@UlrichEckhardt thanks a lot for bringing this issue up. Would you mind opening a pull request with the proposed changes? That would provide a better way to review and discuss the changes.

**Comment by UlrichEckhardt** [comment]
Okay, the first PR had a failure in a test (the test itself seems to have failed, not the tested code), but now it should be ready. Any questions left, @sagikazarmark ?

**Comment by sagikazarmark** [comment]
Fixed in #505
